contracts/libraries/RMM01Lib.sol

```diff
@@ -72,12 +72,12 @@ library RMM01Lib {
 
         // Uses data.invariant, data.liquidity, and data.remainder to compute next input reserve.
         // Uses next input reserve to compute output reserve.
-        (uint256 prevDep, uint256 nextDep) =
+        (uint256 prevDepTotalWad, uint256 nextDepTotalWad) =
             computeSwapStep(self, data, sellAsset, tau);
 
         // Checks to make sure next reserve decreases and computes the difference in WAD.
-        if (nextDep > prevDep) revert SwapInputTooSmall();
-        data.output += (prevDep - nextDep).mulWadDown(data.liquidity);
+        if (nextDepTotalWad > prevDepTotalWad) revert SwapInputTooSmall();
+        data.output += prevDepTotalWad - nextDepTotalWad;
```

Comment:
No longer divides output by liquidity.

Consequence: `data.output` now is expected to be computed with the dependent reserves which are always scaled from per liquidity amounts to the total amount by multiplying by the pool's liquidity.